flake.nix

```diff
@@ -0,0 +1,109 @@
+{
+  description = "A Nix flake for OSRD dev shell";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
```

Comment:
The stuff from osrd-nix-config also works with stable, which I'd prefer we use unless we really need stuff from unstable

Reply:
clippy is outdated on latest stable, and reports false positives.